powershell/Tests/L0/Write-LoggingCommand.WritesCommand.ps1

```diff
@@ -62,53 +85,48 @@ Invoke-VstsTaskScript -ScriptBlock {
     try {
         foreach ($variableSet in $variableSets) {
             $command = $variableSet.Command
-            Unregister-Mock Write-Host
-            Register-Mock Write-Host
 
-            # Act/Assert - Verify using as output switch.
+            # Verify using as output switch.
+            # Act.
             $actual = & $vstsModule Write-LoggingCommand @command -AsOutput
-            Assert-WasCalled Write-Host -Times 0
+            # Assert.
             if (!$variableSet.AlternateExpected -or $variableSet.AlternateExpected -ne $actual) {
                 Assert-AreEqual $variableSet.Expected $actual
             }
 
-            # Act/Assert - Verify using "Object" parameter set and as output switch.
+            # Verify using "Object" parameterset and as output switch.
+            # Act.
             $actual = & $vstsModule Write-LoggingCommand -Command $command -AsOutput
-            Assert-WasCalled Write-Host -Times 0
+            # Assert.
             if (!$variableSet.AlternateExpected -or $variableSet.AlternateExpected -ne $actual) {
                 Assert-AreEqual $variableSet.Expected $actual
             }
 
-            # Act/Assert - Verify without using as output switch.
+            # Verify without using as output switch.
+            # Arrange.
+            # Override Write-Host for as short a time as possible. Otherwise the test output becomes confusing.
+            $script:hostMessages = @( )
+            Register-Mock Write-Host { $OFS = " " ; $script:hostMessages += "$args" }
```

Comment:
I can't avoid mocking write-host in this function since this test is actually asserting the behavior of the private helper function that either writes the logging command to the host or to the output pipeline.

I changed this test so that Write-Host is overridden for as short a time as possible. Overriding it for the full test made the test difficult to debug.

I also added a helper assert method to help out with the unpredictable-order-of-the-logging-command-properties issue and other difficulties asserting write-host was called with the expected value.